contracts/upgradeables/chips/Chips.sol

```diff
@@ -142,11 +141,12 @@ contract Chips is
         _grantRole(DEV_CONFIG_ROLE, _devWallet);
         _grantRole(MANAGER_ROLE, _devWallet);
         _setRoleAdmin(MANAGER_ROLE, DEV_CONFIG_ROLE);
-        _setRoleAdmin(GAME_ROLE, MANAGER_ROLE);
         _setRoleAdmin(READABLE_ROLE, MANAGER_ROLE);
 
         _addWhitelistSigner(_devWallet);
         token = IERC20(_token);
+        treasury = _treasury;

```

Comment:
I feel that the treasury it's fine, just that the transfer in each payout() function might be too expensive in mainnet or another chain.